libs/ECS/src/SceneManager.cpp

```diff
@@ -0,0 +1,35 @@
+//
+// Created by Daan van Donk on 10/05/2022.
+//
+
+#include <future>
+#include "SceneManager.h"
+#include "Component.h"
+
+void UpdatePoll(Scene scene);
+
+void SceneManager::LoadScene(Scene scene)
+{
+    for (auto gameObject: scene.gameobjects)
+    {
+        for (auto* component: gameObject.components)
+        {
+            component->Awake();
+        }
+    }
+
+     auto f = std::async(std::launch::async, UpdatePoll, scene);
```

Comment:
Waarom wordt hier async gebruikt? Het lijkt een fire and forget. Hoe werkt async in C++?

Reply:
Ik wilde hem async hebben omdat de UpdatePoll best intensive kan worden en dus niet alle processing in één keer opeet en de rest van het programma stopt. Hoe async werkt in C++? Ik ging er vanuit hetzelfde als C# en Java

Reply:
Is misschien goed voor future proofing, maar ik zie een paar MulitThread problemen ontstaan en extra complexiteit voor nu. Zolang performance nog geen probleem is, lekker niet doen